internal/pithosctl/pkg/pithos/keys.go

```diff
@@ -59,7 +59,7 @@ func generateKeyAndSecret() (key cluster.KeyString, secret cluster.KeyString, er
 }
 
 func randomHex(length int) (cluster.KeyString, error) {
-	data := make([]byte, 32)
+	data := make([]byte, 32, 32*2)
```

Comment:
I meant you need to cap `length` at `32*2` to avoid out of bounds:
```go
func randomHex(length int) (cluster.KeyString, error) {
	const maxLength = 32 << 1
	if length > maxLength {
		length = maxLength
	}
	data := make([]byte, 32)
	_, err := rand.Read(data)
	if err != nil {
		return "", trace.Wrap(err)
	}
	return cluster.KeyString(fmt.Sprintf("%x", sha256.Sum256(data))[:length]), nil
}
```